service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java

```diff
@@ -1638,11 +1571,29 @@ private HttpRequest CreateRequest(URL url, HttpMethod method, byte[] payload, St
         HttpRequest request = new HttpRequest(url, method, payload, proxy);
         request.setReadTimeoutMillis(options.getHttpReadTimeout());
         request.setConnectTimeoutMillis(options.getHttpConnectTimeout());
-        request.setHeaderField("authorization", sasToken);
+
+        request.setHeaderField("authorization", authenticationToken);
         request.setHeaderField("Request-Id", "1001");
         request.setHeaderField("Accept", "application/json");
         request.setHeaderField("Content-Type", "application/json");
         request.setHeaderField("charset", "utf-8");
         return request;
     }
+
+    private String getAuthenticationToken()
+    {
+        // Three different constructor types for this class, and each type provides either a TokenCredential implementation,
+        // an AzureSasCredential instance, or just the connection string. The sas token can be retrieved from the non-null
+        // one of the three options.
+        if (this.credential != null)
+        {
+            return this.credential.getToken(new TokenRequestContext()).block().getToken();
```

Comment:
For this case, you will have to append - "bearer" to the token value and add the expiry of the token.
https://github.com/Azure/azure-sdk-for-net/blob/8970711b5c82c9fac3f7135529ab57de5a294f08/sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs#L218
It is also worth sending an email to the service team to confirm if this is what they expect. I have not gotten this far in code so had not sent an email.